internal/pkg/template/workload.go

```diff
@@ -456,6 +491,17 @@ type NetworkLoadBalancer struct {
 	CertificateRequired bool
 }
 
+// ApplicationLoadBalancer holds configuration that's needed for an Application Load Balancer.
+type ApplicationLoadBalancer struct {
+	PublicSubnetCIDRs []string
+	Listener          []ApplicationLoadBalancerRoutineRule
+	HostedZoneAliases AliasesForHostedZone
+	HTTPRedirect      bool
+	HTTPSListener     bool
```

Comment:
nit: consider renaming to
```suggestion
	RedirectToHTTPS      bool // Only relevant if IsHTTPS is true.
	IsHTTPS     bool  // True if the listener listening on port 443.
```
so that it reads in the template `{{$listener.IsHTTPS}}` or `{{$listener.RedirectToHTTPS}}`